x-pack/legacy/plugins/code/server/routes/lsp.ts

```diff
@@ -208,6 +184,88 @@ export function lspRoute(
       }
     },
   });
+
+  async function groupFiles(req: RequestFacade, list: Location[]) {
```

Comment:
nit: these 2 functions look more like util functions that can be moved out of the `route` file and have some minor unit test cases covered in case someone breaks it.